MAPL_Base/sun.H

```diff
@@ -131,30 +165,80 @@
                MAPL_SunSummerSolstice   ) 
 
             FAC = real(SEC_OF_DAY)/86400.
-            ANG = 2.0*MAPL_PI*FAC
 
             if(TIME_==0) then
                IDAYP1 = mod(IDAY,ORBIT%DAYS_PER_CYCLE) + 1
 
-               ZS  = ORBIT%ZS(IDAYP1)*FAC + ORBIT%ZS(IDAY)*(1.-FAC)
-               ZC  = ORBIT%ZC(IDAYP1)*FAC + ORBIT%ZC(IDAY)*(1.-FAC)
-               AA  = ORBIT%PP(IDAYP1)*FAC + ORBIT%PP(IDAY)*(1.-FAC)
+               ZS = ORBIT%ZS(IDAYP1)*FAC + ORBIT%ZS(IDAY)*(1.-FAC)
+               ZC = ORBIT%ZC(IDAYP1)*FAC + ORBIT%ZC(IDAY)*(1.-FAC)
+               AA = ORBIT%PP(IDAYP1)*FAC + ORBIT%PP(IDAY)*(1.-FAC)
+               if (apply_EOT) &
+                 ET = ORBIT%ET(IDAYP1)*FAC + ORBIT%ET(IDAY)*(1.-FAC)
             else
                call GETIDAY(IDAY,TIME_,ORBIT,RC=STATUS)
                VERIFY_(STATUS)
 
-               ZS  = ORBIT%ZS(IDAY)
-               ZC  = ORBIT%ZC(IDAY)
-               AA  = ORBIT%PP(IDAY)
+               ZS = ORBIT%ZS(IDAY)
+               ZC = ORBIT%ZC(IDAY)
+               AA = ORBIT%PP(IDAY)
+               if (apply_EOT) ET = ORBIT%ET(IDAY)
             endif
 
-            ZTH =         ZS*SIN(LATS) +       ZC*COS(LATS) &
-                 * (sin(ANG)*SIN(LONS) - cos(ANG)*COS(LONS))
+            ! Greenwich MEAN solar hour angle OFFSET by PI
+            ! (since FAC is zero at mignight)
+            ANG = 2.0*MAPL_PI*FAC
+
+            ! apply equation of time correction?
+            if (apply_EOT) then
+
+              ! the real (zero at noon) Greenwich MEAN solar hour angle
+              ANG = ANG + MAPL_PI
+
+              ! Greenwich TRUE solar hour angle
+              ANG = ANG + ET
+
+              ! LOCAL solar zenith angle
+              ZTH = ZS*SIN(LATS) + ZC*COS(LATS)*COS(ANG+LONS)
+
+            else
+
+              ! the historical GEOS-5 calculation based on mean solar time,
+              ! i.e., lacking the required equation of time correction
+
+              ! pmn: this branch can eventually go if EOT becomes default
+              ! pmn: the sin*sin-cos*cos is less clear and probably less
+              !      efficient than the EOT branch anyway
+
+              ! solar zenith angle (based on MEAN solar time)
+              ZTH = ZS*SIN(LATS) + &
+                    ZC*COS(LATS) * (sin(ANG)*SIN(LONS) - cos(ANG)*COS(LONS))
+
+            end if
+
+            ! enforce zero insolation for sun below horizon
             ZTH = max(ZTH, 0.0)
+
+            ! normalized downward solar flux at TOA
             SLR = ZTH*AA
 
             if(present(DIST)) DIST = AA
 
+            ! ---%---
+
+! pmn: Andrea said the following fixed cases are for single column tests,
```

Comment:
Might want to put Andrea's full name.   Someday people may pull us out of retirement to ask us why something was done in such-and-such manner.

Reply:
Added a change in the log, and added Andrea's name as long as I was in there.